packages/apps/helpdesk/src/screens/TicketFormScreen.js

```diff
@@ -16,267 +16,110 @@
-import {
-  ContactPartnerSearchBar,
-  CustomerSearchBar,
-  DurationFormInput,
-  ProgressFormInput,
-  ProjectSearchBar,
-  TicketTypeSearchBar,
-  UserSearchBar,
-} from '../components';
-import {Ticket} from '../types/';
-
-const PROJECT_KEY = 'project';
-const CUSTOMER_KEY = 'customerPartner';
-const CONTACT_KEY = 'contactPartner';
-
-const DEFAULT_TICKET_VALUE = {duration: 0};
-
-const isObjectMissingRequiredField = object => checkNullString(object?.subject);
+import {getCustomerbyId} from '../features/customerSlice';
+import {Ticket} from '../types';
 
 const TicketFormScreen = ({navigation, route}) => {
   const idTicket = route?.params?.idTicket;
-  const dispatch = useDispatch();
+
+  const _dispatch = useDispatch();
   const I18n = useTranslator();
-  const Colors = useThemeColor();
 
   const {ticket} = useSelector(state => state.ticket);
 
-  const [_ticket, setTicket] = useState(
-    idTicket != null ? ticket : DEFAULT_TICKET_VALUE,
-  );
-  const [disabledButton, setDisabledButton] = useState(
-    isObjectMissingRequiredField(_ticket),
-  );
-
-  const handleTicketFieldChange = useCallback((newValue, fieldName) => {
-    setTicket(current => {
-      if (current?.[fieldName] === newValue) {
-        return current;
-      }
-
-      const updatedObject = current == null ? {} : {...current};
-
-      updatedObject[fieldName] = newValue;
-
-      if (fieldName === PROJECT_KEY) {
-        updatedObject[CUSTOMER_KEY] = newValue?.clientPartner;
-        updatedObject[CONTACT_KEY] = newValue?.contactPartner;
-      }
-
-      setDisabledButton(isObjectMissingRequiredField(updatedObject));
-
-      return updatedObject;
-    });
-  }, []);
-
-  const updateTicketAPI = useCallback(() => {
+  const defaultValue = useMemo(() => {
+    const _default = {
+      duration: 0,
+    };
+
+    if (idTicket != null) {
+      return {
+        ...ticket,
+        duration: ticket.duration || _default.duration,
+        prioritySelect: {
+          title: Ticket.getPriority(ticket?.prioritySelect, I18n),
+          key: ticket.prioritySelect,
+        },
+      };
+    }
+
+    return _default;
+  }, [idTicket, ticket, I18n]);
+
+  const createTicketAPI = useCallback((_ticket, dispatch) => {
+    const dataToSend = {
+      ..._ticket,
+      prioritySelect: _ticket?.prioritySelect?.key,
+    };
     dispatch(
-      updateTicket({
-        ticket: _ticket,
+      createTicket({
+        ticket: dataToSend,
       }),
     );
+  }, []);
 
-    navigation.navigate('TicketDetailsScreen', {
-      idTicket: _ticket.id,
-    });
-  }, [dispatch, _ticket, navigation]);
-
-  const createTicketAPI = useCallback(() => {
-    dispatch(
-      createTicket({
+  const updateTicketAPI = useCallback(
+    (_ticket, dispatch) => {
+      const dataToSend = {
+        ..._ticket,
+        prioritySelect: _ticket?.prioritySelect?.key,
         ticket: _ticket,
-      }),
-    );
+        ticketId: _ticket?.id,
```

Comment:
What's the purpose of 
- having all content of ticket in dataToSend but also in dataToSend.ticket ??
- having ticketId in dataToSend, dataToSend.ticket, dataToSend.ticketId AND also in data given to updateTicket ??